docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+can be useful for testing.
+
+Install miniforge and conda
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you do not already have ``conda`` installed, `download and install miniforge
+<https://github.com/conda-forge/miniforge/blob/main/README.md>`_. The details depend on
+your system but the end result is to provide a ``conda`` executable that you can use
+to create and manage isolated Python environments.
+
+Now create and activate an ``astropy-dev`` conda environment using the following::
+
+   conda create -n astropy-dev python graphviz
+   conda activate astropy-dev
+
+Note the ``graphviz`` package is required for building the documentation.
+
+Install the development version of astropy
+------------------------------------------
+
+Now you can install the development version of astropy into your new environment. This
+will install the latest version of astropy from your local git repo, along with
+all the dependencies needed to build and fully test astropy::
+
+   python -m pip install --editable ".[dev_all]"
+
+**Checking the build**
+
+At this point you should be able to import astropy from your locally built version::
+
+   python
+   >>> import astropy
+   >>> astropy.__version__  # note: the exact output will differ
+   '7.0.0.dev303+gb394fda545.d20240613'
+
+Next you may want to try running some or all of the ``astropy`` unit tests.
+Running the full test suite can take a while, so you may want to start with a subset
+of only the coordinates tests::
+
+.. code-block:: shell
+
+   pytest astropy/coordinates
+   pytest
+
+Details on running and writing tests can be found in the :ref:`testing-guidelines`
+section.
+
+.. _contributing.pre-commit:
+
+Install pre-commit
+------------------
+
+This is optional, but *highly recommended* if you will contribute code. Pre-commit is a
+tool that runs a number of :ref:`Continuous Integration (CI) <contributing.ci>` checks
+(e.g. code formatting) on your code before you commit it. If you skip this step then it
+is likely that one or more of those CI checks will fail when you make a pull request,
+resulting in lost time (yours and CI resources).
+
+Installation is straightforward. From the root of the astropy repository, run::
+
+    pre-commit install
+
+Now all of the styling checks will be
+run each time you commit changes without your needing to run each one manually.
+
+.. tip:: To learn more about pre-commit, see the :ref:`pre-commit` section.
+
+.. _contributing.pull_request:
+
+Creating and submitting a pull request
+======================================
+
+You can contribute bug fixes, new features, and documentation updates by submitting a
+GitHub pull request (PR). This section will guide you through the process. We encourage
+you to `ask for help <https://www.astropy.org/help.html>`_ if you get stuck. The astropy
+community is welcoming and friendly and will help you!
+
+.. _contributing.version_control:
+
+GitHub and Git
```

Comment:
This feels like a bit too much duplication from the start of the page.  I'd skip directly to creating a feature branch

Reply:
Agreed, fixed.